src/call_wrapper.ts

```diff
@@ -44,25 +44,27 @@ class CallWrapper {
         }
     }
 
-    private static calculateInitialMemorySize(parameters: any[]): number {
+    private static calculateInitialMemorySize(parameters: any[], initialMemorySize: number): number {
         const pageSize = Math.pow(2, 16);
         const memoryElementCount = parameters
             .filter((parameter) => parameter instanceof Array)
             .map((array) => array.length + 1)
             .reduce((accumulator, current) => accumulator + current, 0);
 
-        return Math.ceil((memoryElementCount * 4) / pageSize);
+        return Math.ceil(((memoryElementCount + initialMemorySize) * 4) / pageSize);
     }
 
     private readonly hooks: TranspilerHooks;
     private readonly wasmModule: Module;
+    private readonly arrayLiteralMemorySize: number;
 
     private functionName: string;
     private outParameters: any[];
 
-    public constructor(wasmModule: Module, hooks: TranspilerHooks) {
+    public constructor(wasmModule: Module, hooks: TranspilerHooks, arrayLiteralMemorySize: number) {
```

Comment:
There are multiple inconsistent usages of this parameter. I found several names used for the same value:
- arrayLiteralMemorySize
- initialSize
- initialMemorySize

You should decide on one of them.